JavadocStyleGuide.md

```diff
@@ -0,0 +1,223 @@
+ *
+ * @param  url  an absolute URL giving the base location of the image
+ * @param  name the location of the image, relative to the url argument
+ * @return      the image at the specified URL
+ * @see         Image
+ */
+public Image getImage(URL url, String name) {
+   try {
+       return getImage(new URL(url, name));
+   } catch (MalformedURLException e) {
+       return null;
+   }
+}
+```
+
+Basic formatting rules:
+
+- The first line contains the begin-comment delimiter ( `/**`).
+- Notice the inline tag `{@link URL}`, which converts to an HTML hyperlink pointing to the documentation for the URL class.
+- If you have more than one paragraph in the doc comment, separate the paragraphs with a `<p>` paragraph tag, as shown.
+- Insert a blank comment line between the description and the list of tags, as shown.
+- The first line that begins with an `@` character ends the description; you cannot continue the description following block tags.
+- Block tags must be added in order.
+- The last line contains the end-comment delimiter ( `*/`).
+
+So lines won't wrap, limit any doc-comment lines to 80 characters.
+
+### 1.1.2 HTML
+
+Javadoc permits only a subset of HTML tags: `"a", "abbr", "acronym", "address", "area", "b", "bdo", "big", "blockquote", "br", "caption", "cite", "code", "colgroup", "dd", "del", "dfn", "div", "dl", "dt", "em", "fieldset", "font", "h1", "h2", "h3", "h4", "h5", "h6", "hr", "i", "img", "ins", "kbd", "li", "ol", "p", "pre", "q", "samp", "small", "span", "strong", "sub", "sup", "table", "tbody", "td", "tfoot", "th", "thead", "tr", "tt", "u", "ul", "var"`.
+
+Attempt to limit your usage of HTML to: `"h1", "h2", "h3", "h4", "em", "li", "ol", "p", "ul"`.
+
+Some HTML element types allow authors to omit end tags: `<p>` and `<li>`.
+
+### 1.1.3 Paragraphs
+
+One blank line—that is, a line containing only the aligned leading asterisk (`*`)—appears between the last paragraph and before the group of block tags if present:
+
+```java
+/**
+ * that draw the image will incrementally paint on the screen.
+ *
+ * @param  url  an absolute URL giving the base location of the image
+ */
+```
+
+Each paragraph except the first has `<p>` immediately before the first word, with no space after it:
+
+```java
+/**
+ * argument is a specifier that is relative to the url argument.
+ * <p>This method always returns immediately, whether or not the
+ */
+```
+
+Alternatively, a `<p>` can be placed on a separate line:
+
+```java
+/**
+ * argument is a specifier that is relative to the url argument.
+ * <p>
+ * This method always returns immediately, whether or not the
+ */
+```
+
+The inline tag `{@link URL}`, which converts to an HTML hyperlink pointing to the documentation for the URL class. 
+This inline tag can be used anywhere that a comment can be written, such as in the text following block tags:
+
+```java
+/**
+ * The url argument must specify an absolute <a href="#{@link}">{@link URL}</a>. The name
```

Comment:
This document references the `<a href="#{@link}">{@link URL}</a>` syntax a lot, but I hardly see it in practice in the real javadoc in `gradle/gradle`. 

Generally I much more often see `{@link ClassName}` without any `a href` tag. IMO we should focus on that format

Reply:
Styling guide has been updated.